tests/layer_tests/tensorflow_tests/test_tf_MatMul.py

```diff
@@ -0,0 +1,97 @@
+        op_type_to_tf = {
+            'BatchMatMul': tf.raw_ops.BatchMatMul,
+            'BatchMatMulV2': tf.raw_ops.BatchMatMulV2,
+            'BatchMatMulV3': tf.raw_ops.BatchMatMulV3,
+            'MatMul': tf.raw_ops.MatMul,
+        }
+
+        tf.compat.v1.reset_default_graph()
+
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            tf_x_shape = x_shape.copy()
+            tf_y_shape = y_shape.copy()
+
+            tf_x = tf.compat.v1.placeholder(tf.float32, tf_x_shape, 'Input')
+            constant_value = np.random.randint(-3, 3, tf_x_shape).astype(np.float32)
+            tf_y = tf.constant(constant_value)
+            if op_type == 'MatMul':
+                if len(x_shape) > 2 or len(y_shape) > 2:
+                    pytest.skip("MatMul doesn't support rank > 2")
+                op_type_to_tf[op_type](a=tf_x, b=tf_y, transpose_a=x_bool, transpose_b=y_bool, name='Operation')
+            elif op_type == 'BatchMatMulV3':
+                op_type_to_tf[op_type](x=tf_x, y=tf_y, Tout=tf.float32, adj_x=x_bool, adj_y=y_bool, name='Operation')
+            else:
+                op_type_to_tf[op_type](x=tf_x, y=tf_y, adj_x=x_bool, adj_y=y_bool, name='Operation')
+
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        ref_net = None
+
+        return tf_net, ref_net
+
+    test_data_precommit = [
+        dict(x_shape=[4, 4], y_shape=[4, 4]),               #Tests 2D shapes
+        dict(x_shape=[4, 4, 4, 4], y_shape=[4, 4, 4, 4]),   #Tests 4D shapes
+        dict(x_shape=[4, 4], y_shape=[4]),                  #Tests broadcast
+        ]
+
+    @pytest.mark.parametrize("params", test_data_precommit)
+    @pytest.mark.parametrize("op_type", ['BatchMatMul',
+                                         'BatchMatMulV2',
+                                         #'BatchMatMulV3',      #Isn't supported
+                                         'MatMul',
+                                         ])
+    @pytest.mark.precommit_tf_fe
+    def test_matmul_op_precommit(self, params, ie_device, precision, ir_version, temp_dir, op_type,
+                                 use_new_frontend, use_old_api):
+        self._test(*self.create_net_with_matmul_op(**params, ir_version=ir_version, op_type=op_type,
+                                                  use_new_frontend=use_new_frontend, x_bool=False, y_bool=False),
+                   ie_device, precision, ir_version, temp_dir=temp_dir,
+                   use_new_frontend=use_new_frontend, use_old_api=use_old_api)
+
+    test_data = test_data_precommit + [
+        dict(x_shape=[4, 4, 4, 4], y_shape=[4, 4]),         #Tests 4D shapes broadcast
+        ]
+
+    @pytest.mark.parametrize("params", test_data)
+    @pytest.mark.parametrize("op_type", ['BatchMatMul',
+                                         'BatchMatMulV2',
+                                         #'BatchMatMulV3',      #Isn't supported
```

Comment:
please create a ticket for TF FE and mark this case as xfail with the ticket number

Reply:
Will it be correct? I didn't see it in supported list. I added it here because it is pretty similar with others for further enabling.